server/channels/api4/webhook_test.go

```diff
@@ -952,31 +959,36 @@ func TestUpdateIncomingHook(t *testing.T) {
 
 	t.Run("PrivateChannel", func(t *testing.T) {
 		privateChannel := th.CreatePrivateChannel()
-		th.Client.Logout(context.Background())
+		_, err = th.Client.Logout(context.Background())
+		require.NoError(t, err)
 		th.LoginBasic()
 		createdHook.ChannelId = privateChannel.Id
 
-		_, resp, err := th.Client.UpdateIncomingWebhook(context.Background(), createdHook)
+		var resp *model.Response
+		_, resp, err = th.Client.UpdateIncomingWebhook(context.Background(), createdHook)
 		require.Error(t, err)
 		CheckForbiddenStatus(t, resp)
 	})
 
 	th.TestForSystemAdminAndLocal(t, func(t *testing.T, client *model.Client4) {
 		createdHook.ChannelId = "junk"
-		_, resp, err := client.UpdateIncomingWebhook(context.Background(), createdHook)
+		var resp *model.Response
+		_, resp, err = client.UpdateIncomingWebhook(context.Background(), createdHook)
 		require.Error(t, err)
 		CheckNotFoundStatus(t, resp)
 	}, "UpdateToNonExistentChannel")
 
 	team := th.CreateTeamWithClient(th.Client)
 	user := th.CreateUserWithClient(th.Client)
 	th.LinkUserToTeam(user, team)
-	th.Client.Logout(context.Background())
-	th.Client.Login(context.Background(), user.Id, user.Password)
+	_, err = th.Client.Logout(context.Background())
+	require.NoError(t, err)
+	_, _, err = th.Client.Login(context.Background(), user.Username, user.Password)
+	require.NoError(t, err)
 	t.Run("UpdateToADifferentTeam", func(t *testing.T) {
 		_, resp, err := th.Client.UpdateIncomingWebhook(context.Background(), createdHook)
```

Comment:
This now panics:
```
goroutine 2121 [running]:
testing.tRunner.func1.2({0x2dfb220, 0x57a6a60})
	/usr/local/go/src/testing/testing.go:1631 +0x24a
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1634 +0x377
panic({0x2dfb220?, 0x57a6a60?})
	/usr/local/go/src/runtime/panic.go:770 +0x132
github.com/mattermost/mattermost/server/public/model.(*Client4).UpdateIncomingWebhook(0xc004913030, {0x3be3440, 0x5879240}, 0x0)
	/mattermost/server/public/model/client4.go:4991 +0x1ba
github.com/mattermost/mattermost/server/v8/channels/api4.TestUpdateIncomingHook.func19(0xc001d2ba00)
	/mattermost/server/channels/api4/webhook_test.go:989 +0x38
testing.tRunner(0xc001d2ba00, 0xc0028e0588)
	/usr/local/go/src/testing/testing.go:1689 +0xfb
created by testing.(*T).Run in goroutine 541
	/usr/local/go/src/testing/testing.go:1742 +0x39
``
It seems like `createdHook` is `nil`. Can you please check why that is the case?

Reply:
I have debugged and got to know where the error in the test is coming from.
UpdateIncomingWebhook function is returning  404 in response.
When i run this test locally i get
                Error:      	Not equal: 
            	            	expected: 403
            	            	actual  : 404
            	Test:       	TestUpdateIncomingHook/UpdateToADifferentTeam
            	Messages:   	Expected http status:403, got 404
Its because in th.TestForSystemAdminAndLocal(t, func(t *testing.T, client *model.Client4) {
		createdHook.ChannelId = "junk"
		var resp *model.Response
		_, resp, err = client.UpdateIncomingWebhook(context.Background(), createdHook)
		require.Error(t, err)
		CheckNotFoundStatus(t, resp)
	}, "UpdateToNonExistentChannel")  Here we are setting channelId to junk which will basically return 404 as there is no such hookId url. If i commend this out the test is working. @hanzei